interfaces/cython/cantera/onedim.py

```diff
@@ -639,6 +622,28 @@ def E(self):
         Efield.append((phi[np-2] - phi[np-1]) / (z[np-1] - z[np-2]))
         return Efield
 
+    def solve(self, loglevel=1, refine_grid=True, auto=False, stage=1, enable_energy=True):
+        self.flame.set_solvingStage(stage)
+        if stage == 1:
+            super(IonFlameBase, self).solve(loglevel, refine_grid, auto)
+        if stage == 2:
+            self.poisson_enabled = True
+            super(IonFlameBase, self).solve(loglevel, refine_grid, auto)
+
+
+class IonFlame(IonFlameBase, FreeFlame):
```

Comment:
Should this class be named `FreeIonFlame`, or `IonFreeFlame` (also renaming `BurnerIonFlame` to `IonBurnerFlame` for consistency)? I have a mild preference for this latter naming convention.